### PR TITLE
JetBrains: Make it compatible with 2022.2

### PR DESCRIPTION
**Pull request by vdavid**
Fixes https://github.com/sourcegraph/sourcegraph/issues/39571

- Updated platform+Java+IntelliJ versions
- Fixed the theme bug: apparently, JetBrains started using a new color class
- Bumped plugin version to 2.0.0, to go ahead with the release

## Test plan

Tested it, it looks nice again. When changing the theme, the “panda bug” reappears, but it's not directly related to this bug. I've created a separate issue for this, with priority:high, but in the Icebox: https://github.com/sourcegraph/sourcegraph/issues/39574

## App preview:

- [Web](https://sg-web-dv-jetbrains-2022-2-compatibility.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-jwfngpyduk.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java [hunk 1]
```diff
@@ -26,7 +27,7 @@ public static JsonObject getCurrentThemeAsJson() {
         for (Object key : keysList) {
             try {
                 Object value = UIManager.get(key);
-                if (value instanceof ColorUIResource) {
+                if (value instanceof ColorUIResource || value instanceof LafManagerImpl.IJColorUIResource) {
```
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java:30]: 🤦

File: client/jetbrains/gradle.properties [hunk 2]
```diff
@@ -3,7 +3,7 @@
 pluginGroup=com.sourcegraph.jetbrains
 pluginName=Sourcegraph
 # SemVer format -> https://semver.org
-pluginVersion=2.0.0-alpha.5
+pluginVersion=2.0.0
```
Comment (philipp-spiess) [inline client/jetbrains/gradle.properties:6]: 🚢

**Review by philipp-spiess — APPROVED** [review]
We also need to update the CHANGELOG I believe 🙂

File: client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java (hunk 1 above)
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java:30]: Does `LafManagerImpl.IJColorUIResource` exist on lower IntelliJ versions? I.e have you tried running the extension on 2021.1?

Comment (philipp-spiess) [reply client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java:30]: image

Oh no

**Comment by vdavid** [comment]
> We also need to update the CHANGELOG I believe 🙂

CHANGELOG is always a pain in the ass for me because I never know the format. I've changed `[Unreleased]` to `[2.0.0]` and added some details. Do you know if this is the right format @philipp-spiess?

File: client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java [hunk 3]
```diff
@@ -26,7 +25,7 @@ public static JsonObject getCurrentThemeAsJson() {
         for (Object key : keysList) {
             try {
                 Object value = UIManager.get(key);
-                if (value instanceof ColorUIResource) {
+                if (value instanceof Color) {
```
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java:28]: Great find!

**Review by philipp-spiess — APPROVED** [review]
Ship it!

Verified that it works on 2021.1 on Windows:

Screenshot 2022-07-28 at 15 41 12

**Pull request merged**